components/raftstore/src/store/peer.rs

```diff
@@ -187,14 +187,20 @@ pub struct CheckTickResult {
 }
 
 pub struct ProposedAdminCmd<S: Snapshot> {
+    cmd_type: AdminCmdType,
     epoch_state: AdminCmdEpochState,
```

Comment:
Seems `epoch_state` is not more necessary, instead of keep the `epoch_state` on `ProposedAdminCmd `, we can get the `AdminCmdEpochState` from `ADMIN_CMD_EPOCH_MAP` with `cmd_type` whenever we need it

Reply:
Yes. But the memory cost is very little because the max proposed admin cmd size, for now, is 2.